### PR TITLE
Move debug print statements to logger and enable user provided logger

### PR DESCRIPTION
**Pull request by timabrmsn**
This change moves `py42.settings.debug` from a module to a class so we don't have to break the existing API for setting debug level. It keeps the existing defined levels as constants, but just translates those to the `logging` module's log levels. So our existing instructions still work:

`py42.settings.debug.level = py42.settings.debug.DEBUG`

But we should probably change our documentation to just use normal logging module levels:

`py42.settings.debug.level = logging.DEBUG`

The only difference in behavior is the removal of TRACE as a real level (I'm currently just equating it to DEBUG for compatibility), prior to this request headers would only print at TRACE, but now they print at DEBUG (since there is no lower level than DEBUG in the `logging` module). 

As for users providing their own logger, all that's needed is:

`py42.settings.debug.logger = my_logger`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 The intended audience of this file is for py42 consumers -- as such, changes that don't affect
 how a consumer would use the library (e.g. adding unit tests, updating documentation, etc) are not captured here.
 
+## Unreleased
+
+### Added
+
+-
```
Comment (alanag13) [inline CHANGELOG.md:15]: Be sure to complete this, and add a note re: `TRACE` being removed

File: src/py42/settings/__init__.py [hunk 2]
```diff
@@ -23,3 +24,24 @@ def get_user_agent_string():
...
+    def __init__(self):
+        self.logger = logging.getLogger("py42")
+        self.logger.addHandler(logging.StreamHandler(sys.stderr))
+
+    @property
+    def level(self):
+        return self.logger.level
+
+    @level.setter
+    def level(self, level):
```
[the remark this answers is not in the document]
Comment (alanag13) [reply src/py42/settings/__init__.py:44]: The python standard library doesn't expose any functions like that, so I actually think that would be extraneous / unexpected for a lot of developers. Writing a convenience method for your own application to do something like that (like if we wanted to add that to the CLI) if you wanted to would be simple enough that I don't think we need to necessarily do it.

**Comment by alanag13** [comment]
@timabrmsn can you look at the codcov diff for `settings/__init__.py` and cover those few straggling cases, ideally let's try not to decrease coverage on any files.

Also, can we add docs showing how to use this (if they don't exist already) and how to supply your own custom logger?

File: src/py42/settings/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply src/py42/settings/__init__.py:44]: `py42.settings.debug.level = py42.settings.debug` is not that simple and looks bad imo

[the remark this answers is not in the document]
Comment (unparalleled-js) [reply src/py42/settings/__init__.py:44]: `py42.settings.set_log_level_to_debug()` doesn't require having to look this up. It doesn't require poking around the settings module. It doesn't require such a repetitious statement

[the remark this answers is not in the document]
Comment (unparalleled-js) [reply src/py42/settings/__init__.py:44]: `py42.settings.debug.level = logging.DEBUG` is do-able I guess

[the remark this answers is not in the document]
Comment (alanag13) [reply src/py42/settings/__init__.py:44]: This change would encourage the norm to be to use the `logging` constants instead, so:
`py42.settings.debug.level = logging.DEBUG`.
The py42 ones are there for backwards compatibility.

Adding a method similar to what the standard library has (like `py42.settings.debug.set_level(logging.DEBUG)`) could be nice though, how do you feel about that?

[the remark this answers is not in the document]
Comment (timabrmsn) [reply src/py42/settings/__init__.py:44]: yeah, `py42.settings.debug.level = py42.settings.debug.DEBUG` still works but is only left possible for backwards compatibility, I did change the documentation examples to use `logging.DEBUG` instead. 

I'm fine with adding a `set_level()` method.

**Pull request merged**